config/hubs/schema.yaml

```diff
@@ -74,6 +75,38 @@ properties:
       - project
       - cluster
       - zone
+  aws:
+    type: object
+    additionalProperties: false
+    description: |
+      Configuration to connect to the cluster on AWS. Is used when
+      provider is set to `aws`
+    properties:
+      key:
+        type: string
+        description: |
+          Path to a `sops` encrypted JSON key that contains AWS credentials
+          with enough rights to get full access to the kubernetes cluster.
```

Comment:
I'm not so used to AWS credentials and I have not yet worked with a .json file to contain them.

Are these to contain the credentials to an account with enough rights to get full access, or is it more indirect like an account that contains credentials to take on a AWS Role, and then the role has credentials etc?

Could you be a bit more specific about this?

Reply:
I will be more specific in the description, of course.
In this case, we have created a `deployer` user in the carbonplan AWS account with enough credentials to interact and work with the EKS cluster (the deployer user belongs to a group with full admin access).
In the future, we may want to revisit this approach and work with AWS roles so we have better granularity and easiness at the time to add users (such as the deployer one) to the EKS cluster (without the mapping we currently need to do). I tried this approach first because it is the most straightforward and because none of our current docs or practices in AWS land includes AWS roles... yet.